dev-docs/RFCs/v4.0.0/names_for_concepts_types_functions.md

```diff
@@ -0,0 +1,142 @@
+
+We should clarify in the documentation that H3 works with **topological** hexagons and pentagons, rather than true **geometric** hexagons and pentagons.
+
+See, for example: https://github.com/uber/h3-js/issues/53
+
+
+## General Function Names
+
+|          Current name         |     Proposed name     |
+|-------------------------------|-----------------------|
+| *Does Not Exist (DNE)*        | `isValidIndex`        |
+| `h3IsValid`                   | `isValidCell`         |
+| `h3UnidirectionalEdgeIsValid` | `isValidDirectedEdge` |
+| `h3IsPentagon`                | `isPentagon`          |
+| `h3ToParent`                  | `cellParent`          |
+| `h3ToChildren`                | `cellChildren`        |
+| `h3IndexesAreNeighbors`       | `cellsAreNeighbors`   |
+| `geoToH3`                     | `geoToCell`           |
+| `h3ToGeo`                     | `cellToGeo`           |
+| `numHexagons`                 | `numCells`            |
+| `getRes0Indexes`              | `getRes0Cells`        |
+| `getPentagonIndexes`          | `getPentagons`        |
+| `h3GetBaseCell`               | `getBaseCellNumber`   |
+
+### naming note
+
+- `isValid*` should mean that a full validity check is made
+- without `Valid` (like in the case of `isPentagon`), we do not guarantee
+  that a full validity check is made; instead, a user should assume only a
+  minimal bit mask check is done
+    - we could imagine a `isValidPengaton` function, if full verification
+      would be convenient
+- similarly, a function like `cellsAreNeighbors` will assume each cell
+  has passed the `isValidCell` check; the function will do only minimal
+  work to determine if they are neighbors
+
+### todo
+- Do we have general guidance on when to use a `get` prefix with a function and when not?
+
+
+## H3 Grid Functions
+
+| Current name |     Proposed name      |
+|--------------|------------------------|
+| `h3Distance` | `gridDistance`         |
+| `h3Line`     | `gridPathCell`         |
+| *DNE*        | `gridPathEdge`         |
+| *DNE*        | `gridPathDirectedEdge` |
+
+
+## H3 Edge Types
+
+Instead of `UnidirectionalEdge`, use the term `DirectedEdge`.
+
+For a future undirected edge mode, use the term `Edge`.
+
+Function name changes:
+
+|                  Current name                 |        Proposed name         |
+|-----------------------------------------------|------------------------------|
+| `getH3UnidirectionalEdge`                     | `getDirectedEdge`            |
+| `h3UnidirectionalEdgeIsValid`                 | `isValidDirectedEdge`        |
+| `getOriginH3IndexFromUnidirectionalEdge`      | `getDirectedEdgeOrigin`      |
+| `getDestinationH3IndexFromUnidirectionalEdge` | `getDirectedEdgeDestination` |
+| `getH3IndexesFromUnidirectionalEdge`          | `getDirectedEdgeCells`       |
+| `getH3UnidirectionalEdgesFromHexagon`         | `getDirectedEdgesFromCell`   |
+| `getH3UnidirectionalEdgeBoundary`             | `getDirectedEdgeBoundary`    |
```

Comment:
As per above, `get` seems inconsistent here. Why `cellToGeoBoundary` but not `directedEdgeToGeoBoundary`? (Also, if we think `GeoBoundary` is too long, we should standardize on `Boundary`.)

Since we're spending so much time on naming, it would be good to have a guideline for when to use a `get-` prefix and when not to - e.g.
- Use `get` when retrieving constant data
- Use `get` when retrieving a subset of an object, or metadata about the object (e.g. `getEdgeOrigin`, `getCellResolution`)
- Use `to` when transforming between different representations of the same object, or when doing a lossy transformation to a new object

In this case, that might mean `cellToDirectedEdges` and `directedEdgeToCells` and possible `cellsToDirectedEdge`. Just spitballing to some extent, but it would be good to have a consistent line of reasoning for using `get` and `to`.

Reply:
I think we've resolved this. Let me know if you see any issues with the latest version!